### PR TITLE
[wasm] WasmAppHost docs

### PR DESCRIPTION
**Pull request by maraf**
Describe how to configure and use WasmAppHost

### CODE REVIEW

**Comment by maraf** [comment]
@radical Any suggestion for a better location?

**Comment by radical** [comment]
> @radical Any suggestion for a better location?

I think we can have it in `src/mono/wasm/host/README.md`. And then link it from a top level md in `docs/`, and `src/mono/wasm/README.md`.

**Comment by maraf** [comment]
Moved to `src/mono/wasm/README.md`. 
The `docs/README.md` doesn't have anything wasm related, so I didn't add it there.

File: src/mono/wasm/README.md [hunk 1]
```diff
@@ -175,6 +175,10 @@ To build and run the samples with AOT, add `/p:RunAOTCompilation=true` to the ab
 
 Also check [bench](../sample/wasm/browser-bench/README.md) sample to measure mono/wasm runtime performance.
 
+## Was App Host
```
Comment (radical) [inline src/mono/wasm/README.md:178]:
```suggestion
## Wasm App Host
```

File: src/mono/wasm/host/README.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+- `perHostConfig`: An array of configuration per host type.
+- `defaultConfig`: A name of the default per-host configuration.
+- `firefoxProxyPort`: A port number where Mono debug proxy for Firefox is listening.
+- `firefoxDebuggingPort`: A port number where Firefox is listening for remote debugging.
+- `chromeProxyPort`: A port number where Mono debug proxy for Chrome is listening.
+- `chromeDebuggingPort`: A port number where Chrome is listening for remote debugging.
+
+## Per host configuration
+
+Wasm App Host supports running the application on various hosts, like browser, node js, v8, etc. These hosts has configuration in the `wasmHostProperties/perHostConfig`. Each of the configuration in this array has a name that can be passed as a command-line argument `--host` (`-h`) when starting the application, or as a `defaultConfig` in the `runtimeconfig.template.json`. If it's not specified, a first one declaration is used.
```
Comment (radical) [inline src/mono/wasm/host/README.md:25]:
```suggestion
Wasm App Host supports running the application on various hosts, like browser, node js, v8, etc. These hosts has configuration in the `wasmHostProperties/perHostConfig`. Each of the configurations in this array has a name that can be passed as a command-line argument `--host` (`-h`) when starting the application, or set as a `defaultConfig` in the `runtimeconfig.template.json`. If it's not specified, the first declaration is used.
```

File: src/mono/wasm/host/README.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
+# Wasm App Host
+
+WasmAppHost is used when `dotnet run` executes for wasm targeting projects.
```
Comment (radical) [inline src/mono/wasm/host/README.md:3]:
```suggestion
WasmAppHost is used when `dotnet run` executes for projects targeting wasm.
```

File: src/mono/wasm/host/README.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+- `chromeProxyPort`: A port number where Mono debug proxy for Chrome is listening.
+- `chromeDebuggingPort`: A port number where Chrome is listening for remote debugging.
+
+## Per host configuration
+
+Wasm App Host supports running the application on various hosts, like browser, node js, v8, etc. These hosts has configuration in the `wasmHostProperties/perHostConfig`. Each of the configuration in this array has a name that can be passed as a command-line argument `--host` (`-h`) when starting the application, or as a `defaultConfig` in the `runtimeconfig.template.json`. If it's not specified, a first one declaration is used. 
+
+> To use JavaScript engines or node, the executable needs to be present in the `PATH`.
+
+Depending on the host type, various properties can be set.
```
Comment (radical) [inline src/mono/wasm/host/README.md:29]:
```suggestion
Depending on the host type, various properties can be set.
```

File: src/mono/wasm/host/README.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+- `html-path`: A relative path of an HTML file to open in the browser. Eg.: `index.html`
+
+### V8
+
+- `host: v8`: A identify this configuration to run on the V8.
+- `js-path`: A relative path of a JavaScript file to execute. Eg.: `main.mjs`
+
+### Node JS
+
+- `host: v8`: A identify this configuration to run on the node.
```
Comment (radical) [inline src/mono/wasm/host/README.md:43]:
```suggestion
- `host: nodejs`: specifies that this configuration is for running with `NodeJS`.
```

File: src/mono/wasm/host/README.md [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+Wasm App Host supports running the application on various hosts, like browser, node js, v8, etc. These hosts has configuration in the `wasmHostProperties/perHostConfig`. Each of the configuration in this array has a name that can be passed as a command-line argument `--host` (`-h`) when starting the application, or as a `defaultConfig` in the `runtimeconfig.template.json`. If it's not specified, a first one declaration is used. 
+
+> To use JavaScript engines or node, the executable needs to be present in the `PATH`.
+
+Depending on the host type, various properties can be set. 
+
+### Browser
+
+- `host: browser`: A identify this configuration to run on the browser.
```
Comment (radical) [inline src/mono/wasm/host/README.md:33]:
```suggestion
- `host: browser`: specifies that this configuration is for running on the browser.
```

File: src/mono/wasm/host/README.md [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+- `html-path`: A relative path of an HTML file to open in the browser. Eg.: `index.html`
+
+### V8
+
+- `host: v8`: A identify this configuration to run on the V8.
```
Comment (radical) [inline src/mono/wasm/host/README.md:38]:
```suggestion
- `host: v8`: specifies that this configuration is for running with `V8`.
```

File: src/mono/wasm/host/README.md [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+- `js-path`: A relative path of a JavaScript file to execute. Eg.: `main.mjs`
+
+### Node JS
+
+- `host: v8`: A identify this configuration to run on the node.
+- `js-path`: A relative path of a JavaScript file to execute. Eg.: `main.mjs`
+
+### SpiderMonkey
+
+- `host: spidermonkey`: A identify this configuration to run on the SpiderMonkey.
```
Comment (radical) [inline src/mono/wasm/host/README.md:48]:
```suggestion
- `host: spidermonkey`: specifies that this configuration is for running with `SpiderMonkey`.
```

File: src/mono/wasm/host/README.md [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+- `js-path`: A relative path of a JavaScript file to execute. Eg.: `main.mjs`
+
+### JavaScriptCore
+
+- `host: javascriptcore`: A identify this configuration to run on the JavaScriptCore.
```
Comment (radical) [inline src/mono/wasm/host/README.md:53]:
```suggestion
- `host: javascriptcore`: specifies that this configuration is for running with `JavaScriptCore`.
```

**Review by radical — APPROVED** [review]
Other than the feedback, LGTM 👍  Thank you for adding this!

**Pull request merged**